File: src/main/java/com/paragon/client/shader/shaders/LiquidShader.java
package com.paragon.client.shader.shaders;

import com.paragon.client.shader.Shader;

import java.awt.*;

import static org.lwjgl.opengl.GL20.*;

public class LiquidShader extends Shader {

    private Color colour = new Color(0, 0, 0);

    public LiquidShader() {
        super("/assets/paragon/glsl/shaders/liquid.frag");
    }

    public void setColour(Color colour) {
        this.colour = colour;
    }

    @Override
    public void setupUniforms() {
        setupUniform("texture");
        setupUniform("resolution");
        setupUniform("time");
        setupUniform("colour");
    }

    @Override
    public void updateUniforms() {
        glUniform1i(getUniform("texture"), 0);
        glUniform2f(getUniform("resolution"), 1F / mc.displayWidth, 1F / mc.displayHeight);
        glUniform1f(getUniform("time"), (float) getTime());
        glUniform4f(getUniform("colour"), colour.getRed() / 255F, colour.getGreen() / 255F, colour.getBlue() / 255F, colour.getAlpha() / 255F);
    }

}

File: src/main/kotlin/com/paragon/client/shader/shaders/LiquidShader.kt
package com.paragon.client.shader.shaders

import com.paragon.client.shader.Shader
import org.lwjgl.opengl.GL20.*
import java.awt.Color

class LiquidShader : Shader("/assets/paragon/glsl/shaders/liquid.frag") {

    private var colour = Color(0, 0, 0)

    fun setColour(colour: Color) {
        this.colour = colour
    }

    override fun setupUniforms() {
        setupUniform("texture")
        setupUniform("resolution")
        setupUniform("time")
        setupUniform("colour")
    }

    override fun updateUniforms() {
        glUniform1i(getUniform("texture"), 0)
        glUniform2f(getUniform("resolution"), 1f / minecraft.displayWidth, 1f / minecraft.displayHeight)
        glUniform1f(getUniform("time"), time.toFloat())
        glUniform4f(getUniform("colour"), colour.red / 255f, colour.green / 255f, colour.blue / 255f, colour.alpha / 255f)
    }

}